client/src/document/index.tsx

```diff
@@ -132,6 +132,36 @@ export function Document(props /* TODO: define a TS interface for this */) {
     }
   }, []);
 
+  React.useEffect(() => {
+    if (!doc) {
+      return;
+    }
+    const handleClick = (event) => {
+      if (event.target instanceof HTMLAnchorElement) {
+        gaSendOnNextPage([
+          {
+            hitType: "event",
+            eventCategory: "Code Block Reference Link",
+            eventAction: event.target.textContent,
+          },
+        ]);
+      }
+    };
+    const removeListeners = Array.from(
+      document.querySelectorAll(".code-example")
+    ).map((codeBlock) => {
+      codeBlock.addEventListener("click", handleClick);
+      return () => {
+        codeBlock.removeEventListener("click", handleClick);
+      };
+    });
+    return () => {
+      for (const removeListener of removeListeners) {
+        removeListener();
+      }
+    };
+  }, [doc, ga]);
```

Comment:
FYI, I know this PR is just an experiment but this approach is most likely not performant. It's much better to set up a onclick event listener on the `pre code` and then within the 1 event listener callback you look at the `event.target` instead.

Reply:
That was just my thinking, which is why I attached it to `.code-example`, which is the parent element of these `pre` elements. It was just a simpler to select class than `.brush` which has many variations in content.

Reply:
SILLY ME! I read the code wrong in a rush. This loop was for each `pre` block! Not each `a` in each `pre` block. /me shuts up.